File: tests/pywin_client-py2many/win32com_/test/testAXScript.jl
module testAXScript
using PyCall
win32api = pyimport("win32api")
using win32com.test.util: RegisterPythonServer

import win32com.axscript
import win32com.axscript.client

import win32com.test.util
abstract type AbstractAXScript <: Abstractwin32com.test.util.TestCase end
verbose = "-v" in append!([PROGRAM_FILE], ARGS)
mutable struct AXScript <: AbstractAXScript
    verbose
end
function setUp(self::AXScript)
    file = GetFullPathName(win32api, join)
    self.verbose = verbose
    RegisterPythonServer(file, "python", self.verbose)
end

function testHost(self::AXScript)
    file = GetFullPathName(win32api, join)
    cmd = "%s \"%s\"" % (GetModuleFileName(win32api, 0), file)
    if verbose
        println("Testing Python Scripting host")
    end
    ExecuteShellCommand(win32com.test.util, cmd, self)
end

function testCScript(self::AXScript)
    file = GetFullPathName(win32api, join)
    cmd = "cscript.exe \"%s\"" % file
    if verbose
        println("Testing Windows Scripting host with Python script")
    end
    ExecuteShellCommand(win32com.test.util, cmd, self)
end

function main()
    testmain(win32com.test.util)
end

main()
end

File: tests/pywin/win32com_/test/testAXScript.py
# Test AXScripting the best we can in an automated fashion...
import win32api, os, sys

import win32com_.axscript
import win32com_.axscript.client

import unittest
import win32com_.test.util

verbose = "-v" in sys.argv


class AXScript(win32com_.test.util.TestCase):
    def setUp(self):
        file = win32api.GetFullPathName(
            os.path.join(win32com_.axscript.client.__path__[0], "pyscript.py")
        )
        from win32com_.test.util import RegisterPythonServer

        self.verbose = verbose
        RegisterPythonServer(file, "python", verbose=self.verbose)

    def testHost(self):
        file = win32api.GetFullPathName(
            os.path.join(win32com_.axscript.__path__[0], "test\\testHost.py")
        )
        cmd = '%s "%s"' % (win32api.GetModuleFileName(0), file)
        if verbose:
            print("Testing Python Scripting host")
        win32com_.test.util.ExecuteShellCommand(cmd, self)

    def testCScript(self):
        file = win32api.GetFullPathName(
            os.path.join(win32com_.axscript.__path__[0], "Demos\\Client\\wsh\\test.pys")
        )
        cmd = 'cscript.exe "%s"' % (file)
        if verbose:
            print("Testing Windows Scripting host with Python script")
        win32com_.test.util.ExecuteShellCommand(cmd, self)


if __name__ == "__main__":
    win32com_.test.util.testmain()
